akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -314,15 +333,33 @@ private[projection] class R2dbcOffsetStore(
   }
 
   private def readTimestampOffset(): Future[Option[TimestampOffset]] = {
-    idle.set(false)
+    implicit val sys = system // for implicit stream materializer
+    triggerDeletionPerSlice.clear()
     val oldState = state.get()
 
-    dao.readTimestampOffset().map { recordsWithKey =>
+    val offsetSliceReadParallelism = 10 // FIXME config
+
+    val recordsWithKeyFut =
+      Source(minSlice to maxSlice)
+        .mapAsyncUnordered(offsetSliceReadParallelism) { slice =>
+          dao.readTimestampOffset(slice)
```

Comment:
We need to read offsets to find the start offset, and also good to pre-populate the state with latest offsets for each slice. Rest will be lazy loaded when needed.